build/scripts/CrossBranchPorting.psm1

```diff
@@ -21,81 +21,92 @@ function New-BCAppsBackport() {
+                $existingPr = $existingOpenPullRequests | Where-Object { ($_.title -eq $title) -and ($_.baseRefName -eq $TargetBranch) }
+                if ($existingPr) {
+                    Write-Host "Pull request for $TargetBranch already exists: $($existingPr.url)" -ForegroundColor Yellow
+                    continue
+                }
+
+                # Create a new branch for the cherry-pick
+                $cherryPickBranch = "backport/$TargetBranch/$branchNameSuffix"
+
+                # Port the pull request to the target branch
+                PortPullRequest -PullRequestDetails $pullRequestDetails -TargetBranch $TargetBranch -CherryPickBranch $cherryPickBranch
+
+                # Create a pull request for the cherry-pick
+                gh pr create --title $title --body $body --base $TargetBranch --head $cherryPickBranch
+
+                $backportPr = gh pr view $cherryPickBranch --json url | ConvertFrom-Json
+                if ($backportPr.url) {
+                    $pullRequests.Add($TargetBranch, $backportPr.url)
+                }
             }
         } catch {
-            RunAndCheck git checkout $startingBranch
             Write-Host "Failed to backport to $TargetBranch. Please inspect the error and try again." -ForegroundColor Red
             Write-Host $_.Exception.Message -ForegroundColor Red
             throw $_.Exception.Message
+        } finally {
+            # Go back to the original branch
+            RunAndCheck git checkout $startingBranch
         }
-    }
 
-    # Go back to the original branch
-    RunAndCheck git checkout $startingBranch
+        if ($pullRequests.Count -eq 0) {
+            Write-Host "No pull requests created" -ForegroundColor Yellow
+            return
+        }
 
-    if ($pullRequests.Count -eq 0) {
-        Write-Host "No pull requests created" -ForegroundColor Yellow
-        return
+        Write-Host "Backport pull requests created:" -ForegroundColor Green
+        foreach($pullRequest in $pullRequests.GetEnumerator()) {
+            Write-Host " - $($pullRequest.Key): $($pullRequest.Value)" -ForegroundColor Green
+        }
+    } catch {
+        Write-Host $_ -ForegroundColor Red
     }
-
-    Write-Host "Backport pull requests created:" -ForegroundColor Green
-    foreach($pullRequest in $pullRequests.GetEnumerator()) {
-        Write-Host " - $($pullRequest.Key): $($pullRequest.Value)" -ForegroundColor Green
+    finally {
+        Pop-Location
     }
 }
 
 function PrecheckBackport($TargetBranches, $PullRequestNumber) {
     # Check gh cli is installed
     if (-not (Get-Command gh -ErrorAction SilentlyContinue)) {
-        throw "Please install the GitHub CLI from https://cli.github.com/"
+        Write-Host "Please install the GitHub CLI by running 'winget install --id GitHub.cli' from your terminal. Once installed open a new terminal and run 'gh auth login' to authenticate with GitHub." -ForegroundColor Red
```

Comment:
Is it possible to just run the command if the user gives consent?
Then says open a new terminal and run `gh auth login`. If that's more work, just leave it

Reply:
Considering the user has to close and reopen their shell, I'll leave running the winget command as a manual step for now. 

About `gh auth login`; It is probably a good idea to validate that gh cli is authenticated (and authenticate if it isn't) prior to starting the backport. I've added a check for the PrecheckBackport function